### PR TITLE
Expose that `BasicQueue` expects blocking spawn

### PR DESCRIPTION
**Pull request by bkchr**
Up to now `BasicQueue` expected a closure that to spawn a `Future`.
This was expected to be a closure that spawns a blocking future.
However, this wasn't documented anywhere. This pr introduces a new trait
`SpawnBlocking` that exposes this requirement to the outside.

polkadot-companion: https://github.com/paritytech/polkadot/pull/1061

### CODE REVIEW

File: primitives/consensus/common/src/import_queue/basic_queue.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ impl<B: BlockT, Transaction: Send + 'static> BasicQueue<B, Transaction> {
 			finality_proof_import,
 		);
 
-		spawner(future.boxed());
+		spawner.spawn_blocking("basic-block-import-worker", future.boxed());
```
Comment (andresilva) [inline primitives/consensus/common/src/import_queue/basic_queue.rs:70]: @tomaka might need to update his grafana dashboard :)

File: primitives/core/src/traits.rs [hunk 2]
```diff
@@ -330,3 +330,11 @@ impl TaskExecutorExt {
 		Self(spawn_handle)
 	}
 }
+
+/// Something that can spawn a blocking future.
+pub trait SpawnBlocking: futures::task::Spawn {
```
Comment (tomaka) [inline primitives/core/src/traits.rs:335]:
```suggestion
pub trait SpawnBlocking {
```

**Comment by tomaka** [comment]
What about changing the closure to accept an additional `TaskType` parameter instead? That would be consistent with the API of `sc_service`.
When I see a new trait being introduced, I can't help but be annoyed. This adds a lot of complexity for not much.

**Comment by DemiMarie-parity** [comment]
I agree with @tomaka on this.

**Comment by bkchr** [comment]
The point why I come up with this, because I needed to port the changes to Cumulus. If I would just have read the documentation, I would probably just have spawned the future without using `spawn_blocking`. However I looked into the code and copy pasted the code, but this can not be the way we write code. The interface should express this and not require some opaque closure that takes a future.

I don't understand how forwarding random closures can help to solve anything? Especially as this random closure directly sets an opioniated name for the future. What happens when I refactor the BasicQueue and need to spawn another future? Right, I would use the closure and spawn another future using the same name. This increases the spaghettification of the code. 

I would like to have used a upstream trait as well, but there is nothing.

**Pull request merged**